source/lac/petsc_matrix_base.cc

```diff
@@ -489,6 +489,87 @@ namespace PETScWrappers
   }
 
 
+  namespace internals
+  {
+    void perform_mmult (const MatrixBase &inputleft,
+                        const MatrixBase &inputright,
+                        MatrixBase       &result,
+                        const VectorBase &V,
+                        const bool        transpose_left)
+    {
+      const bool use_vector = (V.size() == inputright.m() ? true : false);
+      if (transpose_left == false)
+        {
+          Assert (inputleft.n() == inputright.m(),
+                  ExcDimensionMismatch(inputleft.n(), inputright.m()));
+        }
+      else
+        {
+          Assert (inputleft.m() == inputright.m(),
+                  ExcDimensionMismatch(inputleft.m(), inputright.m()));
+        }
+
+      result.clear();
+
+      PetscErrorCode ierr;
+
+      if (use_vector == false)
+        {
+          if (transpose_left)
+            {
+              ierr = MatTransposeMatMult (inputleft,
```

Comment:
You are missing the `Assert` statements here and in the `else` branch to test whether `ierr` is zero.

Reply:
Fixed.